src/library/tools/R/check.R

```diff
@@ -3925,15 +3925,15 @@ add_dummies <- function(dir, Log)
 
     check_rust <- function()
     {
-        ## A Cargo.toml file is used to identify a rust package, 
+        ## A Cargo.toml file is used to identify a rust package,
         ## so check for Cargo.toml under src to see if rust code
         ## is compiled in the crate
         if (is.na(InstLog)) return (NA)
         srcd <- file.path(pkgdir, "src")
-        if (length(list.files(path = srcd, pattern = "Cargo.toml", recursive = TRUE))) == 0) return (NA)
+        if (length(list.files(path = srcd, pattern = "Cargo.toml", recursive = TRUE)) == 0) return (NA)
         ##message("InstLog = ", InstLog)
         lines <- readLines(InstLog, warn = FALSE)
-        l1 <- grep("(cargo build|   Compiling )", lines)
+        l1 <- grep(r"(cargo\N+build|   Compiling )", lines)
```

Comment:
Typo?

```suggestion
        l1 <- grep(r"(cargo\s+build|   Compiling )", lines)
```

Reply:
no: wanted to also capture configuration and toolchain use if CRAN lets that happen in future.

all the following are valid calls:
`cargo build`
`cargo   build`
`cargo +1.71 build` (builds with specifically Rust version 1.71)
`cargo   +stable  build` (builds with current stable version)
`cargo cargo --config profile.dev.package.image.opt-level=3 build` (overrides a global config setting)

Reply:
In the submission of the patch or in a comment I would mention this change allows to catch more cases of packages building rust. I am not sure R Core team or Brian Ripely is aware of this possibility.

But I was checking regex, and I only found that \N is:

> The backreference ‘⁠\N⁠’, where ‘⁠N = 1 ... 9⁠’, matches the substring previously matched by the Nth parenthesized subexpression of the regular expression. (This is an extension for extended regular expressions: POSIX defines them only for basic ones.)

And as this backreference is inside the parenthesis I am not sure how it will work. That's why I proposed a change. When I want to match arbitrary characters between two known sets I usually use .+ because according to the documentation "The period ‘⁠.⁠’ matches any single character.".

Reply:
Looks like it's intended to mean the [anything but a newline](https://pcre.org/current/doc/html/pcre2pattern.html#SEC7) signifier in PCRE2. It will need `perl = TRUE` to work.